cmake/host-tools.cmake

```diff
@@ -7,6 +7,24 @@ find_program(
   )
 if(${WEST} STREQUAL WEST-NOTFOUND)
   unset(WEST)
+else()
+  # Detect whether we are inside a west installation by parsing the output
+  # of west --version, which outputs both the bootstrapper and repository
+  # versions when inside one
+  execute_process(
+    COMMAND
+    ${WEST} --version
+    OUTPUT_VARIABLE west_version_output
```

Comment:
The alternative approach can be found here: https://github.com/tejlmand/zephyr/commit/0fd80529edceb4122ccafb8d36e63e607b6943e6

Reply:
>I would like to suggest an alternative, which is to simply do the check in west list and if that fails, then there are no modules to import as that indicates an mono-repo installation.

We discussed this offline yesterday with @mbolivar. The issue is that `west list` can fail for other reasons, such as a malformed manifest. That makes it then confusing for the user if west is indeed correctly inside an installation but the manifest has some mistaken in it.

Reply:
I still think changing the approach makes sense.

In future `west` may be valid for other commands, even if `west list` can fail in a mono-repo install, example can be `west config <name>` that can fetch general `west` settings, similar to `git config` can be used outside a git repo. So completely disabling `west` may not be that desired, as `west` is indeed installed.

But I acknowledge that `west list` can fail for other reasons, and when that happens the message should still be printed. Thus I updated the proposal here: https://github.com/tejlmand/zephyr/commit/b92ec0d4dae6c6210932a13e63083c5094b914e8 so that in case the failure is not because we are outside a `west` managed project, then the error is still printed.